File: suitecrm-migration/backend/template-service/src/main/java/com/suitecrm/template/service/TemplateService.java
package com.suitecrm.template.service;

import com.suitecrm.template.dto.*;
import com.suitecrm.template.entity.*;
import com.suitecrm.template.repository.*;
import lombok.RequiredArgsConstructor;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.stereotype.Service;
import java.util.UUID;

@Service @RequiredArgsConstructor
public class TemplateService {
    private final EmailTemplateRepository emailTemplateRepository;
    private final PdfTemplateRepository pdfTemplateRepository;

    public Page<EmailTemplateDto> getAllEmailTemplates(Pageable pageable) {
        return emailTemplateRepository.findByDeletedFalse(pageable).map(this::toEmailTemplateDto);
    }
    public EmailTemplateDto getEmailTemplateById(UUID id) {
        return toEmailTemplateDto(emailTemplateRepository.findByIdAndDeletedFalse(id)
                .orElseThrow(() -> new RuntimeException("Email template not found: " + id)));
    }
    public Page<PdfTemplateDto> getAllPdfTemplates(Pageable pageable) {
        return pdfTemplateRepository.findByDeletedFalse(pageable).map(this::toPdfTemplateDto);
    }
    public PdfTemplateDto getPdfTemplateById(UUID id) {
        return toPdfTemplateDto(pdfTemplateRepository.findByIdAndDeletedFalse(id)
                .orElseThrow(() -> new RuntimeException("PDF template not found: " + id)));
    }

    private EmailTemplateDto toEmailTemplateDto(EmailTemplate t) {
        return EmailTemplateDto.builder().id(t.getId()).name(t.getName()).subject(t.getSubject())
                .body(t.getBody()).bodyHtml(t.getBodyHtml()).type(t.getType()).textOnly(t.getTextOnly())
                .description(t.getDescription()).assignedUserId(t.getAssignedUserId())
                .dateEntered(t.getDateEntered()).dateModified(t.getDateModified()).build();
    }
    private PdfTemplateDto toPdfTemplateDto(PdfTemplate t) {
        return PdfTemplateDto.builder().id(t.getId()).name(t.getName()).type(t.getType())
                .moduleName(t.getModuleName()).body(t.getBody()).pdfHeader(t.getPdfHeader())
                .pdfFooter(t.getPdfFooter()).pageSize(t.getPageSize()).orientation(t.getOrientation())
                .description(t.getDescription()).assignedUserId(t.getAssignedUserId())
                .dateEntered(t.getDateEntered()).dateModified(t.getDateModified()).build();
    }
}

File: suitecrm-migration/frontend/shell/src/app/services/template.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

export interface EmailTemplate {
  id: string;
  name: string;
  subject: string;
  body: string;
  bodyHtml: string;
  type: string;
  textOnly: boolean;
}

export interface PdfTemplate {
  id: string;
  name: string;
  type: string;
  moduleName: string;
  body: string;
  pageSize: string;
  orientation: string;
}

@Injectable({ providedIn: 'root' })
export class TemplateService {
  private apiUrl = '/api/templates';

  constructor(private http: HttpClient) {}

  getEmailTemplates(page = 0, size = 20): Observable<any> { return this.http.get(`${this.apiUrl}/email?page=${page}&size=${size}`); }
  getEmailTemplateById(id: string): Observable<EmailTemplate> { return this.http.get<EmailTemplate>(`${this.apiUrl}/email/${id}`); }
  getPdfTemplates(page = 0, size = 20): Observable<any> { return this.http.get(`${this.apiUrl}/pdf?page=${page}&size=${size}`); }
  getPdfTemplateById(id: string): Observable<PdfTemplate> { return this.http.get<PdfTemplate>(`${this.apiUrl}/pdf/${id}`); }
}
